lighthouse/utils/runner.py

```diff
@@ -0,0 +1,166 @@
+"""
+Utility functions for running workloads.
+"""
+
+import numpy as np
+import os
+from mlir import ir
+from mlir.dialects import func, arith, scf, memref
+from mlir.execution_engine import ExecutionEngine
+from mlir.runtime.np_to_memref import get_ranked_memref_descriptor
+from lighthouse.utils.mlir import get_mlir_library_path
+from lighthouse.utils import memrefs_to_packed_args
+from lighthouse import Workload
+from typing import Optional
+
+
+def get_engine(
+    payload_module: ir.Module, shared_libs: list[str] = None, opt_level: int = 3
+) -> ExecutionEngine:
+    lib_dir = get_mlir_library_path()
+    libs = []
+    for so_file in shared_libs or []:
+        so_path = os.path.join(lib_dir, so_file)
+        if not os.path.isfile(so_path):
+            raise ValueError(f"Could not find shared library {so_path}")
+        libs.append(so_path)
+    execution_engine = ExecutionEngine(
+        payload_module, opt_level=opt_level, shared_libs=libs
+    )
+    execution_engine.initialize()
+    return execution_engine
+
+
+def execute(
```

Comment:
Could we move both `execute` and other functions taking `Workload` to `lighthouse.workload` (creating a folder if so needed)?

To me these functions seem closely tied to the `Workload` interface (and have a bit of an opinionated way of interacting with schedules, e.g. how `schedule_parameters` are passed).

Reply:
workload.py and runner.py have now been moved under lighthouse/workload